generated/nidaqmx/nidaqmx_service.cpp

```diff
@@ -14001,15 +14001,15 @@ namespace nidaqmx_grpc {
     using CallbackRouter = nidevice_grpc::CallbackRouter<int32, TaskHandle, int32>;
     class RegisterDoneEventReactor : public nidevice_grpc::ServerWriterReactor<RegisterDoneEventResponse, nidevice_grpc::CallbackRegistration> {
     public:
-    RegisterDoneEventReactor(const RegisterDoneEventRequest& request, NiDAQmxLibraryInterface* library, const ResourceRepositorySharedPtr& session_repository)
+    RegisterDoneEventReactor(::grpc::CallbackServerContext* context, const RegisterDoneEventRequest* request, NiDAQmxLibraryInterface* library, NiDAQmxService* service)
```

Comment:
Regarding service object lifetime:
- The library and service objects are managed with the same shared_ptr: https://github.com/ni/grpc-device/blob/main/generated/nidaqmx/nidaqmx_service_registrar.cpp . If the service object can be deleted out from under us, then so can the library object.
- The constructor and `start()` method are passed the resource repository shared_ptr by reference, not by value. The reference is to a member variable of the service class.